crates/pipeline_manager/src/db/storage.rs

```diff
@@ -441,6 +443,74 @@ pub(crate) trait Storage {
     /// Returns error if there does not exist a service with the provided name.
     async fn delete_service(&self, tenant_id: TenantId, service_name: &str) -> Result<(), DBError>;
 
+    /// Creates a new service probe. Because in the creation of probes, the
```

Comment:
You needn't describe why this API is transactional, because it might not be the only context in which this API is used transactionally.

Reply:
Definitely, will update description, it was left over from when the transaction was non-optional.